File: mobile/src/main/java/com/chaos/databinding/activities/BaseActivity.java
package com.chaos.databinding.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        int layoutId = onCreateViewId();
        this.setContentView(layoutId);
    }

    @Override
    public void setContentView(int layoutResID) {
        super.setContentView(layoutResID);
        //ButterKnife.inject(this);
    }

    protected abstract int onCreateViewId();
}

File: mobile/src/main/kotlin/io/ashdavies/databinding/common/BaseActivity.kt
package io.ashdavies.databinding.common

import android.os.Bundle
import androidx.annotation.IdRes
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatActivity

internal abstract class BaseActivity : AppCompatActivity() {

  @get:LayoutRes
  protected abstract val layoutResId: Int

  @get:IdRes
  protected abstract val toolbarId: Int

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    setContentView(layoutResId)
    setSupportActionBar(findViewById(toolbarId))
  }
}
